test/api/detection_test.cpp

```diff
@@ -38,24 +82,225 @@ TEST(junction_detection, cigar_string_method_simple)
+                                                                                                  {4, 'M'_cigar_operation},
+                                                                                                  {6, 'S'_cigar_operation}}};
+    AlignedSegment aligned_segment6 {strand::forward, "chr1", 150, 60, std::vector<seqan3::cigar>{{44, 'S'_cigar_operation},
+                                                                                                  {6, 'M'_cigar_operation}}};
+    std::vector<AlignedSegment> aligned_segments{aligned_segment1,
+                                                 aligned_segment2,
+                                                 aligned_segment3,
+                                                 aligned_segment4,
+                                                 aligned_segment5,
+                                                 aligned_segment6};
+    std::vector<Junction> junctions_res{};
+    seqan3::dna5_vector query_sequence = {"GGGCTCATCGATCGATTTCGGATCGGGGGGCCCCCATTTTAAACGGCCCC"_dna5};
+    std::string const read_name = "read021";
+    analyze_aligned_segments(aligned_segments,
+                             junctions_res,
+                             query_sequence,
+                             read_name);
+    
+    Breakend new_breakend_1 {"chr1", 106, strand::forward};
+    Breakend new_breakend_2 {"chr2", 100, strand::forward};
+    Breakend new_breakend_3 {"chr1", 106, strand::reverse};
+    Breakend new_breakend_4 {"chr2", 110, strand::reverse};
+    Breakend new_breakend_5 {"chr1", 116, strand::forward};
+    Breakend new_breakend_6 {"chr1", 130, strand::reverse};
+    Breakend new_breakend_7 {"chr1", 116, strand::reverse};
+    Breakend new_breakend_8 {"chr1", 130, strand::forward};
+    Breakend new_breakend_9 {"chr1", 134, strand::forward};
+    Breakend new_breakend_10 {"chr1", 150, strand::forward};
+    std::vector<Junction> junctions_expected_res{Junction{new_breakend_1, new_breakend_2, ""_dna5, read_name},      //translocation
+                                                 Junction{new_breakend_3, new_breakend_4, ""_dna5, read_name},      //translocation
+                                                 Junction{new_breakend_5, new_breakend_6, ""_dna5, read_name},      //inversion
+                                                 Junction{new_breakend_7, new_breakend_8, ""_dna5, read_name},      //inversion
+                                                 Junction{new_breakend_9, new_breakend_10, ""_dna5, read_name}};    //deletion
+
+    ASSERT_EQ(junctions_expected_res.size(), junctions_res.size());
+
+    for (size_t i = 0; i < junctions_expected_res.size(); ++i)
+    {
+        EXPECT_EQ(junctions_expected_res[i].get_read_name(), junctions_res[i].get_read_name());
+        EXPECT_TRUE(junctions_expected_res[i] == junctions_res[i]);
+        // For debugging #include <seqan3/core/debug_stream.hpp> and use:
```

Comment:
Why would I want to include this in debug mode? Could you not include it by checking if the debug mode was used?

Reply:
This has nothing to do with the debug mode of the compiler:
When this test case fails because `junctions_expected_res[i] != junctions_res[i]`, it is sometimes hard to know why. For that case, the comment suggested a few lines of code to investigate whether mate1 or mate2 of the junction were unequal causing the entire junctions to be unequal.

However, it's not ideal that you need to change the code for debugging. Therefore, I now added this debug output as the default error message (using the <<operator). When the test case fails, it will now immediately output which of mate1 or mate2 were unequal.